plugins/node/opentelemetry-instrumentation-aws-lambda/src/instrumentation.ts

```diff
@@ -77,6 +78,18 @@ export class AwsLambdaInstrumentation extends InstrumentationBase {
 
   constructor(protected override _config: AwsLambdaInstrumentationConfig = {}) {
     super('@opentelemetry/instrumentation-aws-lambda', VERSION, _config);
+
+    if (_config.disableAwsContextPropagation == null) {
+      if (
+        typeof env['OTEL_LAMBDA_DISABLE_AWS_CONTEXT_PROPAGATION'] ===
+          'string' &&
+        env[
+          'OTEL_LAMBDA_DISABLE_AWS_CONTEXT_PROPAGATION'
+        ].toLocaleLowerCase() === 'true'
+      ) {
+        _config.disableAwsContextPropagation = true;
```

Comment:
I think this will modify the value of the user's config object, e.g.
```js
const config = { foo: 'bar' };
const instrumentation = new AwsLambdaInstrumentation(config);
console.log(config); // {"foo": "bar", "disableAwsContextPropagation": true }
```

What do you think about copying the config object before modifying it, so that we don't introduce any side-effect to user?

Reply:
think a shallow copy like `_config = { ..._config };` is sufficient or should we deep copy with JSON stringify/destringify and risk crash if the config object contains something not serializable?

Reply:
7f40ea85